docs/ff-concepts/adding-customization/common-examples.md

```diff
@@ -0,0 +1,284 @@
+---
+slug: /concepts/custom-code/common-examples
+title: Common Examples
+description: Learn about the common custom code examples and use it directly in your project.
+tags: [Custom Actions, Custom Code]
+sidebar_position: 6
+keywords: [FlutterFlow, Custom Actions, Customizations, Flutter, Dart, Pub.dev, Examples]
+---
+
+# Common Code Examples
+
+The custom code feature in FlutterFlow allows you to extend functionality by accessing generated classes and modifying global variables like App States and FlutterFlow themes. This guide covers common scenarios where you can leverage custom code to enhance your project by working directly with data models and other resources within your code.
+
+### Accessing FlutterFlow Generated Classes
+
+Most custom FlutterFlow classes are prefixed with `FF<ClassName>` or `FlutterFlow<ClassName>`. If you need to access these classes in your custom code, simply type "FF" or "FlutterFlow" in the code editor to locate them easily.
+
+![suggestions-dropdown.png](imgs/suggestions-dropdown.png)
+
+### Returning Component from Custom Widget 
+
+In a **[Custom Widget](custom-widgets.md)**, you can integrate a previously built **[FlutterFlow Component](../../resources/ui/components/intro-components.md)** directly, saving you from recreating child content in code. For example, if you’re building a Custom Widget to display custom dialog boxes or bottom sheets using a package from 
+[pub.dev](custom-code.md#pubdev), you can simply return an existing Component created on the canvas, rather than coding the content from scratch.
+
+:::tip[Imports]
+When referencing a Component class in your code, FlutterFlow will automatically add the necessary import statement.
+:::
+
+![return-widget-custom-code.png](imgs/return-widget-custom-code.png)
+
+
+### Accessing FlutterFlow Theme in Custom Widget
+
+When building custom widgets, you often need to style parts of the widget, such as setting colors. Instead of using hardcoded color values, you can directly access the **FlutterFlow Theme**. This theme provides consistent styling across your app and reflects colors set by you or your project developer.
+
+To access theme colors in your custom widget, use the `FlutterFlowTheme.of(context)` method. This allows you to retrieve any theme color, such as the **primary color, accent color**, or **background color**, ensuring that your custom widget aligns with the app’s overall theme.
```

Comment:
Call out custom colors as well. And there are other things in theme besides colors, we should reference that too